ipa-core/src/protocol/basics/mul/mod.rs

```diff
@@ -22,35 +22,18 @@ pub(crate) mod malicious;
 mod semi_honest;
 
 #[cfg(feature = "descriptive-gate")]
-pub use semi_honest::multiply as semi_honest_multiply;
+pub use semi_honest::gen_prss_and_multipy as semi_honest_multiply;
 
 /// Trait to multiply secret shares. That requires communication and `multiply` function is async.
 #[async_trait]
 pub trait SecureMul<C: Context>: Send + Sync + Sized {
     /// Multiply and return the result of `a` * `b`.
     async fn multiply<'fut>(&self, rhs: &Self, ctx: C, record_id: RecordId) -> Result<Self, Error>
-    where
-        C: 'fut,
-    {
-        self.multiply_sparse(rhs, ctx, record_id).await
-    }
-
-    /// Multiply and return the result of `a` * `b`.
-    /// This takes a profile of which helpers are expected to send
-    /// in the form (self, left, right).
-    /// This is the implementation you should invoke if you want to
-    /// save work when you have sparse values.
-    async fn multiply_sparse<'fut>(
-        &self,
-        rhs: &Self,
-        ctx: C,
-        record_id: RecordId,
-    ) -> Result<Self, Error>
     where
         C: 'fut;
 }
 
-use semi_honest::multiply as semi_honest_mul;
+use semi_honest::gen_prss_and_multipy as semi_honest_mul;
```

Comment:
maybe we can do away with either this line or line 25 (which is conditional, but not really)